src/main/java/frc/robot/subsystems/swerve/GyroIOPigeon2.java

```diff
@@ -53,8 +55,8 @@ public void updateInputs(GyroIOInputs inputs, List<Samples> asyncOdometrySamples
     inputs.odometryYawPositions =
         asyncOdometrySamples.stream()
             .map(s -> s.values().get(yaw))
-            .map(d -> d == null ? null : Rotation2d.fromDegrees(d))
-            .toArray(Rotation2d[]::new);
+            .map(d -> d == null ? new NullableRotation2d(null) : new NullableRotation2d(new Rotation2d(d)))
```

Comment:
Does this still need to be a Rotation2d.fromDegrees? A static factory on NullableRotation2d might make this a little nicer. `NullableRoation2d.fromDegrees(d)`?